add docker and rufo (#33)
cool-people-make-things/smart-shopper: backend/_GettingStarted.md

rails -v
```

## VSCode Ruby Support

If ruby files look gross in VS Code, install the Ruby extension from Shopify.

## VSCode Ruby Formatter

I've added Rufo as a gem on this project and in combo with the vscode extension it can auto format on save like prettier does on the front end.

You'll need to run the install command to make sure you've got it locally:

```sh
bundle install
```

In VSCode you will need to install the extension by clicking on [this link](vscode:extension/mbessey.vscode-rufo) or search for:

> rufo (Ruby formatter)\
> Matt Bessey

Once gem and extension are installed: in a ruby file, open the context menu and select "Format Document With...". Configure your default formatter as rufo.

## Docker

Docker will be used to run the backend, so you'll need to install it (unless you have it already).

To check if you have it, run:

```sh
docker -v
```

> If you don't have it, you can follow the Docker install guide. I think for Mac and Windows (non WSL) you can use the [Docker Desktop](https://www.docker.com/products/docker-desktop/). But this is what I (a WSL user) did:
> 
> ```sh
> sudo apt-get update
> sudo apt-get install docker.io
> ```
> 
> Then you can create and add your user to a docker group so you can run it without using `sudo` every single time:
> 
> ```sh
> sudo usermod -aG docker $USER
> newgrp docker
> ```
> 

Reviewer: ```suggestion
If you don't have it, you can follow the Docker install guide. 

<details><summary>WSL</summary>
Install Docker

```sh
sudo apt-get update
sudo apt-get install docker.io
```

Then you can create and add your user to a docker group so you can run it without using `sudo` every single time:

```sh
sudo usermod -aG docker $USER
newgrp docker
```

</details> 

<details><summary>Mac</summary>
You can use the [Docker Desktop](https://www.docker.com/products/docker-desktop/)

- Install the Docker Desktop version for you Mac device (Silicon Chip or Intel)
- Follow set up instructions from the application (create account/sign in)

</details>

> 

Reviewer: If you don't have it, you can follow the Docker install guide. 

<details><summary>WSL</summary>
Install Docker

```sh
sudo apt-get update
sudo apt-get install docker.io
```
Then you can create and add your user to a docker group so you can run it without using `sudo` every single time:

```sh
sudo usermod -aG docker $USER
newgrp docker
```
</details> 

<details><summary>Mac</summary>
You can use the [Docker Desktop](https://www.docker.com/products/docker-desktop/)

- Install the Docker Desktop version for you Mac device (Silicon Chip or Intel)
- Follow set up instructions from the application (create account/sign in)

</details>

You can check if it worked by running:

```sh
docker run hello-world
```

And it should show a welcome print out.

## Docker Compose

Docker Compose is used to run multiple containers at once, which is useful for the backend and database stuff (which we'll likely add later).

Again, Mac and Windows can use the Docker Desktop app, and WSL will install it directly.

I, a WSL user, followed the docker docs which talked about certificates and keyrings and stuff, but tbh you can probably just install it directly and it should be fine.

To install it:

```sh
sudo apt-get install docker-compose-plugin
```

> This should let us use the command `docker compose` (with a space) instead of `docker-compose` (with a hyphen).

Reviewer: <details><summary>WSL</summary>
As a WSL user, followed the docker docs which talked about certificates and keyrings and stuff, but tbh you can probably just install it directly and it should be fine.

To install it:

```sh
sudo apt-get install docker-compose-plugin
```

This should let us use the command `docker compose` (with a space) instead of `docker-compose` (with a hyphen).

</details> 

<details><summary>Mac (with Homebrew)</summary>

 Install Homebrew (if you don't already have it) – Get the package manager up and running
 
```sh
 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
 ```
Update brew formulae and cask info – Ensures latest package versions

 ```sh
 brew update  
 ```       
Install Docker Compose – Finally get Compose on your system

 ```sh
 brew install docker-compose  
 ```  
 
And that‘s it! With 3 quick terminal commands, Homebrew pulled down its Docker Compose formula, handled installing any dependencies, resolved conflicts with other apps, and activated the latest supported Docker Compose release.

</details>
